### PR TITLE
[NFC][SYCL][Doc] Improve OpenCL kernel generation doc

### PR DESCRIPTION
**Pull request by Fznamznon**
Signed-off-by: Mariya Podchishchaeva <mariya.podchishchaeva@intel.com>

### CODE REVIEW

**Comment by Fznamznon** [comment]
I worked on preparing a new version of OpenCL kernel generation patch for upstream, found that the commit message was taken from the doc and found that the corresponding doc a little bit ugly. Tried to improve.

File: sycl/doc/SYCLCompilerAndRuntimeDesign.md [hunk 1]
```diff
@@ -102,16 +102,18 @@ pointers to the device memory. As there is no way in OpenCL to pass structures
...
-To facilitate the mapping of the captures/fields of lambdas/functors to OpenCL
-kernel and overcome OpenCL limitations we added the generation of an OpenCL
+the device. In OpenCL kernel arguments are set by calling `clSetKernelArg` function
+for each kernel argument, meanwhile in SYCL all the kernel arguments are fields of
+"SYCL kernel function" which can be defined as a lambda function or a named function
+object and passed as an argument to SYCL function for invoking kernels (such as
+`parallel_for` or `single_task`). For example, in the previous code snippet above
+`accessor` `A` is one such captured kernel argument.
+
+To facilitate the mapping of the fields of SYCL kernel functions to OpenCL
```
Comment (bader) [inline sycl/doc/SYCLCompilerAndRuntimeDesign.md]:
```suggestion
To facilitate the mapping of SYCL kernel data members to OpenCL
```
"fields of .. functions" doesn't make sense. SYCL kernel is not a function it's a function object.

File: sycl/doc/SYCLCompilerAndRuntimeDesign.md [hunk 2]
```diff
@@ -102,16 +102,18 @@ pointers to the device memory. As there is no way in OpenCL to pass structures
...
+kernel arguments and overcome OpenCL limitations we added the generation of an OpenCL
 kernel function inside the compiler. An OpenCL kernel function contains the
-body of the SYCL kernel function, receives OpenCL like parameters and
-additionally does some manipulation to initialize captured lambda/functor fields
+body of the SYCL kernel function, receives OpenCL-like parameters and
+additionally does some manipulation to initialize SYCL kernel function fields
```
Comment (bader) [inline sycl/doc/SYCLCompilerAndRuntimeDesign.md]:
```suggestion
additionally does some manipulation to initialize SYCL kernel data members
```

File: sycl/doc/SYCLCompilerAndRuntimeDesign.md (hunk 1 above)
Comment (Fznamznon) [reply sycl/doc/SYCLCompilerAndRuntimeDesign.md]: It was motivated by the spec. See Glossary section for example.

> SYCL kernel function A type which is callable with ‘operator()‘ that takes a id, item, nd-item or group which can be passed to kernel enqueue member functions of the command group handler. A SYCL kernel function defines an entry point to a kernel. The function object can be a named standard layout type or lambda function.

But you are right, It sounds strange for me too...

**Comment by keryell** [comment]
`[NFC]` seems redundant to me since it is about the `[Doc]`

**Pull request merged**